internal/services/integritymonitor/integritymonitor.go

```diff
@@ -240,11 +248,18 @@ func saveHashes(
 			countHashes++
 		}
 
-		ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
-		defer cancel()
+		//ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+		//defer cancel()
 
-		query, args := data.NewHashFileData().PrepareBatchQuery(hashData, dd)
-		err := repositories.ExecQueryTx(ctx, query, args...)
+		//query, args := data.NewHashFileData().PrepareBatchQuery(hashData, dd)
+		logger.Info("saveHashes - hashData", hashData)
+		err := repositories.NewReleaseStorage(repositories.DB().SQL(), alg, logger).Create(dd)
+		//err := repositories.ExecQueryTx(ctx, query, args...)
+		if err != nil {
+			errC <- err
+		}
+		err = repositories.NewHashStorage(repositories.DB().SQL(), alg, logger).Create(hashData, dd)
```

Comment:
We have 2 tables now but it is stil being a single logical object to store. And we need to write these tables under a single transaction.

Reply:
resolved: created bulkInsert and to write tables under a single transaction. https://github.com/ScienceSoft-Inc/integrity-sum/pull/48/commits/df701c8e8df4c9c3625d41f08958dcbab2fcd4c6